foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/HackingEclipseHelperUtil.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 1998, 2020 Oracle and/or its affiliates. All rights reserved.
+ * This program and the accompanying materials are made available under the 
+ * terms of the Eclipse Public License v1.0 and Eclipse Distribution License v. 1.0 
+ * which accompanies this distribution. 
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at 
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * Contributors:
+ *     Oracle - initial API and implementation from Oracle TopLink
+ ******************************************************************************/  
+package org.eclipse.persistence.internal.helper;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.persistence.config.SystemProperties;
+import org.eclipse.persistence.exceptions.*;
+import org.eclipse.persistence.internal.localization.*;
+import org.eclipse.persistence.internal.identitymaps.CacheKey;
+import org.eclipse.persistence.logging.*;
+
+/**
+ * Helper class to help us build diagnostic strings
+ */
+public class HackingEclipseHelperUtil implements Serializable {
+    
+    public static final HackingEclipseHelperUtil SINGLETON = new HackingEclipseHelperUtil();
+    
+   
+    /**
+     * @return check if the user has specified a system property to control how long we are willing to wait before firing up an exception
+     */
+    public long getMaxAllowedSleepTimeMs() {
+        long defaultMaxAllowedSleepTimeMs = 40000;
+        
+        // (a) Check if a system property was provided by the user to control how long we tolerate before exploding thread waiting to release
+        // deferred locks
+        String eclipseLinkConcurrencyManagerMaxAllowedSleepTimeMs = System.getProperty("eclipseLinkConcurrencyManagerMaxAllowedSleepTimeMs");
```

Comment:
See `PrivilegedAccessHelper.getSystemProperty(...)`. Also, the property name should follow dot (.) notation similar to other properties found in `org.eclipse.persistence.config.SystemProperties` or `org.eclipse.persistence.config.PersistenceUnitProperties`